monkey/tests/unit_tests/common/agent_events/test_hostname_discovery_event.py

```diff
@@ -0,0 +1,76 @@
+from uuid import UUID
+
+import pytest
+
+from common.agent_events import HostnameDiscoveryEvent
+
+AGENT_ID = UUID("012e7238-7b81-4108-8c7f-0787bc3f3c10")
+TIMESTAMP = 1664371327.4067292
+HOSTNAME = "hostname"
+
+HOSTNAME_DISCOVERY_EVENT = HostnameDiscoveryEvent(
+    source=AGENT_ID, timestamp=TIMESTAMP, hostname=HOSTNAME
+)
+
+HOSTNAME_DISCOVERY_OBJECT_DICT = {
+    "source": AGENT_ID,
+    "timestamp": TIMESTAMP,
+    "target": None,
+    "tags": [],
+    "hostname": HOSTNAME,
+}
+
+HOSTNAME_DISCOVERY_SIMPLE_DICT = {
+    "source": str(AGENT_ID),
+    "timestamp": TIMESTAMP,
+    "target": None,
+    "tags": [],
+    "hostname": HOSTNAME,
+}
+
+
+@pytest.mark.parametrize(
+    "event_dict",
+    [HOSTNAME_DISCOVERY_OBJECT_DICT, HOSTNAME_DISCOVERY_SIMPLE_DICT],
+)
+def test_constructor(event_dict):
+    assert HostnameDiscoveryEvent(**event_dict) == HOSTNAME_DISCOVERY_EVENT
+
+
+def test_constructor__tags_are_frozenset():
+    dict_with_tags = HOSTNAME_DISCOVERY_SIMPLE_DICT.copy()
+    dict_with_tags["tags"] = {
+        "tag-1",
+        "tag-2",
+        "tag-3",
+    }
+
+    assert isinstance(HostnameDiscoveryEvent(**dict_with_tags).tags, frozenset)
+
+
+def test_constructor__extra_fields_forbidden():
+    extra_field_dict = HOSTNAME_DISCOVERY_SIMPLE_DICT.copy()
+    extra_field_dict["extra_field"] = 99  # red balloons
+
+    with pytest.raises(ValueError):
+        HostnameDiscoveryEvent(**extra_field_dict)
```

Comment:
Neither of these properties are defines by hostname discovery event. This is testing `AbstractAgentEvent` and `InfectionMonkeyModelConfig`. We can test that, but this is not the place

Reply:
They test that the event inherits `AbstractAgentEvent` and the right model config.

Reply:
You can test that with `issubclass(HostnameDiscoveryEvent, AbstractAgentEvent)`, but I'm not sure we should add these everywhere.